spec/views/hq/partners/_form.html.erb_spec.rb

```diff
@@ -1,25 +1,39 @@
 require 'rails_helper'
 require 'cgi'
 
-RSpec.describe "hq/partners/_form.html.erb", type: :view do
-
+RSpec.describe 'hq/partners/_form.html.erb', type: :view do
   let(:provinces) { Province.all }
   let(:statuses) { Status.all }
   let(:partner) { build_stubbed :partner,
-                  region: "Region1", contact_details: "CD123"}
+                  region: 'Region1', contact_details: 'CD123'}
 
   before :each do
     assign(:provinces, provinces)
     assign(:statuses, statuses)
     assign(:partner, partner)
-    render
   end
 
   specify 'has a form' do
-    assert_select "form"
+    render and assert_select 'form'
   end
 
-  specify 'renders each form value' do
+  describe '"Cancel" button' do
+    specify 'when no :id' do
+      allow(partner).to receive(:id).and_return nil
+      render
+      assert_select 'a[href=?]', hq_partners_path, role: 'button', text: 'Cancel'
+    end
+
+    specify 'when :id' do
+      allow(partner).to receive(:id).and_return 42
+      render
+      assert_select 'a[href=?]', hq_partner_path(42), role: 'button', text: 'Cancel'
+    end
+  end
+
+  specify 'form values' do
+    render
```

Comment:
I like the line between the render and the tests. I'm not bothered which style we use but can we pick one and use it consistently.